app/scripts/components/projectComponents/projectChildren.jsx

```diff
@@ -15,6 +16,8 @@ var mui = require('material-ui'),
 class ProjectChildren extends React.Component {
 
     render() {
+        let uploading = this.props.uploading ? <div><LinearProgress color={"#2196f3"} mode="determinate" value={ProjectStore.uploadProgress} style={styles.uploader}/><div className="mdl-color-text--grey-600" style={styles.uploadText}>uploading...</div></div> : '';
```

Comment:
define uploading as a map over the keys of this.props.uploads. return a LinearProgress for each upload in uploads. include filename in text, i.e. "Uploading file.txt"